### DISCUSSION TITLE
Custom theme colors break dark mode

### DISCUSSION THREAD

**Discussion by RamiAwar** [tailwindlabs/tailwindcss#5111] [Help] [answered]
### Describe the problem:

Adding *global* colors to the theme breaks dark mode.

When adding the colors under extend: {}, this is fixed, but I'm not sure this is expected behavior, as the docs mention that we can define colors without the extend object.

Looking at the generated css, I can see that only dark classes for my new colors are generated. No dark mode classes are generated for anything else like text-white, bg-whatev...

### Link to a minimal reproduction:

https://play.tailwindcss.com/EVmMgwKFom?file=config

**Comment by alexanderbluhm** [comment, marked as answer]
@RamiAwar I am not sure if I understood your problem correctly but when you define colors in the `theme` object, you overwrite **all** existing colors, also black and white. So with your given config, the only colors you have are `light-blue` and `cyan`.
So `dark:text-white` or `dark:bg-red-500` simply makes no sense because there is no `white` or `red` color. Generating these classes would be useless and increase bundle size.
Nevertheless, `dark:text-cyan-600` or `dark:bg-cyan-500` work as expected.

If you want to keep all existing colors and add new ones, you can use `extend`.
If you want to design your very own color palette it is recommended to _overwrite_ the current `theme.colors` property.

Apart from colors this is identical for almost every other tailwind utility like `spacing`, `shadow` etc. You can _extend_ the stuff developed by the Tailwind team or use your own custom `spacing` or `shadow` ...

**Comment by RamiAwar** [comment]
Hey Alexander, yes you totally understood my problem. I didn't think that adding a 'global palette' would overwrite all existing colors. I now understand that it's a feature and not a bug, and now it makes sense to me why extend is there, thanks!

Looking over the documentation again, I think this issue could use some clarification in the colors documentation, as its not very obvious, but feel free to disagree.

You can close this issue!
